packages/legend-studio/src/stores/editor-state/FileGenerationState.ts

```diff
@@ -180,11 +180,10 @@ export class FileGenerationState {
     reprocessOpenNodes(treeData, this.filesIndex, this.root, openedNodeIds);
     // select the current file node if available, else select the first output
     const selectedFileNodePath =
-      this.selectedNode?.fileNode.path ??
-      (generationResult.length === 1
+      generationResult.length === 1
```

Comment:
This is moving in the right direction but we are not reopening the same file in a regenerate that the user had opened instead we are always opening the first element in the generation output. 
I think we should follow the below logic. 
if `this.selectedNode?.fileNode.path` exists and the corresponding file exists in the file index open that. 
If not open the first generationOutput if there is one.